sdk/src/main/java/com/hcaptcha/sdk/HCaptchaDialogFragment.java

```diff
@@ -172,22 +196,18 @@ public void onLoaded() {
         handler.post(new Runnable() {
             @Override
             public void run() {
-                if (showLoader) {
-                    loadingContainer.animate().alpha(0.0f).setDuration(200)
-                            .setListener(
-                                    new AnimatorListenerAdapter() {
-                                        @Override
-                                        public void onAnimationEnd(Animator animation) {
-                                            loadingContainer.setVisibility(View.GONE);
-                                        }
-                                    });
-                } else {
-                    // Add back dialog shadow in case the checkbox or challenge is shown
-                    final Dialog dialog = getDialog();
-                    if (dialog != null) {
-                        dialog.getWindow().addFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);
-                    }
-                }
+                ready();
+            }
+        });
+    }
+
+    @Override
+    public void onOpen() {
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                hCaptchaDialogListener.onOpen();
+                ready();
```

Comment:
Why does `onOpen` need to call `ready`?

Reply:
Fixed, you are right there is no reason to duplicate this logic for `onOpen` and `onLoaded`